pkg/storage/cacher.go

```diff
@@ -627,3 +606,27 @@ func (c *cacheWatcher) process(initEvents []watchCacheEvent, resourceVersion uin
 		}
 	}
 }
+
+type ready struct {
+	ok bool
+	c  *sync.Cond
+}
+
+func newReady() *ready {
+	return &ready{c: sync.NewCond(&sync.Mutex{})}
+}
+
+func (r *ready) wait() {
+	r.c.L.Lock()
+	for !r.ok {
+		r.c.Wait()
+	}
+	r.c.L.Unlock()
+}
+
+func (r *ready) set(ok bool) {
+	r.c.L.Lock()
```

Comment:
I am not sure what do you mean. Do you mean that you worry about someone entries at line 622 without releasing the lock? This is how convar works. Convar will release lock in wait. See https://golang.org/pkg/sync/#Cond.Wait